### PR TITLE
APRL Virtual WAN Recommendations - 1

### PR DESCRIPTION
**Pull request by maurojasmsft**
# Overview/Summary

- Added APRL for Virtual WAN
- Updated Azure Resources regex schema to incorporate P2S @oZakari 

## Related Issues/Work Items

AB#34867

## As part of this pull request I have

- [ x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [ x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [ x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [ ] Ensured PR tests are passing
- [ ] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by oZakari** [comment]
Thanks for the contribution, @maurojasmsft! Could I please have you create a folder in the virtualwans directory called `kql`? You can then add a query file for each recommendation you added with the file name that matches the aprlguid as noted [here](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing/create-content/create-arg-queries/).

Also, above you'll see some of the lint errors. You can click on `Details` to view the changes that need to be made to fix the tests, but if you have any questions please let me know.

**Comment by maurojasmsft** [comment]
@oZakari added folder, files and put // under-development

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (ehaslett) [inline azure-resources/Network/virtualWans/recommendations.yaml:20]: This appears to be more in the area of enabling functionality vs improving resiliency. I am wondering how this would enhance resiliency. In other words, if asymmetrical flows exist, the customer probably is already aware and has taken steps to correct. So what is the benefit in terms of resiliency of turning this on?

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (ehaslett) [inline azure-resources/Network/virtualWans/recommendations.yaml:39]: Similar comment to the previous one. While this may be a good practice, is it focused on improving resiliency, or aligning to architetural best practices?

**Review by ehaslett — CHANGES_REQUESTED** [review]
Please see comments inline for the Routing Intent recommendations and how they promote increased resiliency.

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml:39]: My train of thought here is, enabling the data path across virtual hubs in case of an outage so it doesn't have to traverse through on premises. This would be recommended if:
- Customer has two secure hubs
- Customer is not using custom Route tables
- Customer doesn't have Routing intent enabled.

Let me know your thoughts

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml:20]: When the firewall is non zone redundant and the rack goes down, then there's no resiliency. Adding to different Zones will enhance resiliency in case of Rack going down.

Comment (ehaslett) [reply azure-resources/Network/virtualWans/recommendations.yaml:20]: Thanks - not sure how this showed up but my feedback was around the "Enable routing intent to enable Secured Hubs Interconnectivity". The non zone redundant FW item makes sense, no concern there!

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (ehaslett) [reply azure-resources/Network/virtualWans/recommendations.yaml:39]: My feedback is around line 39 - "Enable routing intent to send private and internet traffic via Azure Firewall or Security Solution deployed in the v-Hub."

**Comment by oZakari** [comment]
@maurojasmsft looks like there is some test errors (ie. duplicate categories, invalid line endings, and a few others). If you open the details of the failed tests you should be able to see all of them. Ping me if you have questions though.
image

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml:20]: Gotcha, my below comment belongs here then

My train of thought here is, enabling the data path across virtual hubs in case of an outage so it doesn't have to traverse through on premises. This would be recommended if:

Customer has two secure hubs
Customer is not using custom Route tables
Customer doesn't have Routing intent enabled.
Let me know your thoughts

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml:39]: This is a good point, I wonder if we can collapse both recommendations. Securing traffic is a requirement to enable routing intent, which will enable that inter hub link communication that otherwise wouldn't be enabled if there's firewalls in each hub.

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (ehaslett) [reply azure-resources/Network/virtualWans/recommendations.yaml:20]: This is an architectural recommendation, and does not necessarily make VWAN itself more resilient in terms of configuration, so my feeling, based on having this discussion with Rodrigo in the past, is that this guidance would be associated with architecture, not specifically VWAN itself.

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (ehaslett) [reply azure-resources/Network/virtualWans/recommendations.yaml:39]: I would also say to this:
This is an architectural recommendation, and does not necessarily make VWAN itself more resilient in terms of configuration, so my feeling, based on having this discussion with Rodrigo in the past, is that this guidance would be associated with architecture, not specifically VWAN itself.

**Review by oZakari — APPROVED** [review]
LGTM, will await approval from @aprl-networking before merging.

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml]: Seems to me I can get rid of 

- description: Enable routing intent to send private and internet traffic via Azure Firewall or Security Solution deployed in the v-Hub.

For this one below:
- description: Enable routing intent to enable Secured Hubs Interconnectivity

Enabling inter hub connectivity wouldn't add to their resiliency as a direct path for inter spoke connectivity? If you don't think this guidance belongs here then I can remove it as well.

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml]: Removed the recommendations pertaining to routing intent

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml]: Removed the recommendations pertaining to routing intent as it falls into the WAF option

File: azure-resources/Network/expressRouteGateways/recommendations.yaml
Comment (ehaslett) [inline azure-resources/Network/expressRouteGateways/recommendations.yaml:1]: The expressRouteGateways section recently was moved/renamed to virtual network gateways, and I think this content would be more appropriate in virtual network gateways. But certainly others can comment on this as well.

Comment (maurojasmsft) [reply azure-resources/Network/expressRouteGateways/recommendations.yaml:1]: vWAN ExR gateway is of a different Resource Provider Microsoft.Network/expressRouteGateways, whereas regular Vnet ExR Gateway is Resource Provider Microsoft.Network/virtualNetworkGateways. Should I still move it where suggested? Microsoft.Network/expressRouteGateways has it's own section hence that's why I put it there

File: azure-resources/Network/virtualWans/recommendations.yaml
Comment (maurojasmsft) [reply azure-resources/Network/virtualWans/recommendations.yaml]: @ehaslett is it ok to resolve this conversation? You happy with this?

File: azure-resources/Network/expressRouteGateways/recommendations.yaml
Comment (ehaslett) [reply azure-resources/Network/expressRouteGateways/recommendations.yaml:1]: I missed that point - sorry. Sounds completely appropriate as is!

**Review by ehaslett — APPROVED** [review]

**Review by ehaslett — APPROVED** [review]
LGTM

**Pull request merged**
